.clang-format

```diff
@@ -0,0 +1,14 @@
+BasedOnStyle: LLVM
+Language: Cpp
+Standard: Cpp11
+IndentWidth: 4
+ColumnLimit: 100
+BinPackParameters: false
+AlwaysBreakTemplateDeclarations: true
+PenaltyReturnTypeOnItsOwnLine: 10000
+BreakConstructorInitializers: AfterColon
+ConstructorInitializerAllOnOneLineOrOnePerLine: true
+PointerAlignment: Left
+
+# Clang version mismatch fixes
+AlignTrailingComments: false
```

Comment:
Check out the update here. It allows us to match ou naming to the one we agreed on for C++
https://github.com/SwarmUS/HiveMind/blob/feat/logging/.clang-tidy

Reply:
Added esp-idf license as well